glean-core/preview/src/core_metrics.rs

```diff
@@ -0,0 +1,113 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use glean_core::{metrics::StringMetric, CommonMetricData, Lifetime};
+
+/// Metrics included in every ping as `client_info`.
+#[derive(Debug)]
+pub struct ClientInfoMetrics {
+    /// The build identifier generated by the CI system (e.g. "1234/A").
+    pub app_build: String,
+    /// The user visible version string (e.g. "1.0.3").
+    pub app_display_version: String,
+    /// The channel the application is being distributed on.
+    pub app_channel: Option<String>,
```

Comment:
For other languages, this is coming as part of the configuration. We can try to be consistent with that as a beginning step, and then see if things need to be broken out differently.

Reply:
Oh right, I forgot that the channel might be passed in.